src/libraries/System.Text.Json/gen/Reflection/TypeExtensions.cs

```diff
@@ -16,6 +16,11 @@ public static string GetCompilableName(this Type type)
                 return GetCompilableName(type.GetElementType()) + "[]";
             }
 
+            if (type.IsGenericParameter)
+            {
+                return type.Name;
```

Comment:
can parameters be also generic? should there be some recursion happening here? (if not, why not? consider adding comment)

Reply:
As far as I know, it cannot. A generic parameter is a placeholder for a type, for example `T` in `List<T>`.
The following code snippet shows that for the inner List in `Lazy<List<T>>` `IsGenericParameter == false`
```
Type propertyType1 = typeof(Test<int>).GetProperty("Prop").PropertyType;
Type argType1 = propertyType1.GetGenericArguments()[0];
Type nestedArgType1 = argType1.GetGenericArguments()[0];
Console.WriteLine($"{argType1.Name} IsGenericParameter: {argType1.IsGenericParameter}");
Console.WriteLine($"{nestedArgType1.Name} IsGenericParameter: {nestedArgType1.IsGenericParameter}");

Type propertyType2 = typeof(Test<>).GetProperty("Prop").PropertyType;
Type argType2 = propertyType2.GetGenericArguments()[0];
Type nestedArgType2 = argType2.GetGenericArguments()[0];
Console.WriteLine($"{argType2.Name} IsGenericParameter: {argType2.IsGenericParameter}");
Console.WriteLine($"{nestedArgType2.Name} IsGenericParameter: {nestedArgType2.IsGenericParameter}");

class Test<T>
{
    public Lazy<List<T>> Prop { get; set; }
}
```
output
```
List`1 IsGenericParameter: False
Int32 IsGenericParameter: False
List`1 IsGenericParameter: False
T IsGenericParameter: True
```

Unfortunately I cannot think of a good comment.